File: 03.programmers/lv03/01.C++/paperfolding.cpp
#include<iostream>
#include <string>
#include <deque>
#include <vector>

using namespace std;
vector<int> solution(int n) {
    vector<int> answer;
    deque<int> tmp;
    vector<int> prev;
    prev.push_back(0);
    for(int i=1; i<n; i++){
        tmp.push_back(0);
        for(int j=prev.size()-1; j>=0; j--){
            tmp.push_front(prev[j]);
            prev[j] = prev[j]==0?1:0;
        }
        for(int j=prev.size()-1; j>=0; j--)
            tmp.push_back(prev[j]);
        prev.clear();
        while(!tmp.empty()){
            prev.push_back(tmp.front());
            tmp.pop_front();
        }
    }
    answer = prev;
    return answer;
}
int main(){
    int n = 3;
    for(int ans : solution(n))
        cout<<ans<<" ";

    return 0;
}

File: ~2023/03.programmers/lv03/02.Python/paperfolding.py
def solution(n):
    answer = []
    for i in range(n):
        tmp = []
        if len(answer) == 0 :
            answer = [0]
            continue
        tmp = answer
        tmp.append(0)  
        for j in range(len(answer)-2,-1,-1):
            if answer[j] == 0 : tmp.append(1)
            else: tmp.append(0)      
        
        answer = tmp
    return answer
input_n = 3
print(solution(input_n))